### PR TITLE
Installer

### PR DESCRIPTION
**Pull request by drewstreib**
Initial install.sh

### CODE REVIEW

File: install.sh [hunk 1]
```diff
@@ -0,0 +1,180 @@
...
+
+You must install cURL to use this installer.
+EOABORT
+  )"
+fi
+
+################### install dmon.py
+
+RAND=${RANDOM}
+curl -s "${DMON_PY_URL}" > /tmp/${RAND}-dmon.py
```
Comment (espadolini) [inline install.sh]: This should use mktemp and it should attempt to trap "rm -f" on exit

Comment (espadolini) [inline install.sh]: Should probably use `-sSf` to display errors and not output anything unless it's a 200.

File: install.sh [hunk 2]
```diff
@@ -0,0 +1,180 @@
...
+/usr/bin/install -D -o "${USER}" -g "${GROUP}" -m "0755" /tmp/${RAND}-dmon.py "${INSTALL_PATH}/dmon.py"
```
Comment (espadolini) [inline install.sh]: Isn't the default to install as current user and current group?

File: install.sh [hunk 3]
```diff
@@ -0,0 +1,180 @@
+#!/bin/bash
+
+# Derived from: 
+# https://raw.githubusercontent.com/Homebrew/install/HEAD/install.sh
+
+# We don't need return codes for "$(command)", only stdout is needed.
+# Allow `[[ -n "$(command)" ]]`, `func "$(command)"`, pipes, etc.
+# shellcheck disable=SC2312
```
Comment (espadolini) [inline install.sh]: Are we actually going to run shellcheck?

File: install.sh [hunk 4]
```diff
@@ -0,0 +1,180 @@
...
+################### install crontab
+
+if [[ ${USER} == "root" ]]
+then
+  # is ok to just overwrite this
+  NEWCRON="* * * * * daemon ${INSTALL_PATH}/dmon.py --net ${NET} --cron ${TELEMETRY_KEY} \"${JOB_NAME}\""
+  echo "${NEWCRON}" > /etc/cron.d/dmon
+else
+  # remove any previous dmon.py entry first, so this automated script only
+  # works for one entry for a user
```
Comment (espadolini) [inline install.sh:151]: I'd add a comment in the crontab line so we are sure to only affect the installer-installed lines.

File: install.sh [hunk 5]
```diff
@@ -0,0 +1,180 @@
...
+rm -f /tmp/${RAND}-dmon.py
+
+
+################### install crontab
+
+if [[ ${USER} == "root" ]]
+then
+  # is ok to just overwrite this
+  NEWCRON="* * * * * daemon ${INSTALL_PATH}/dmon.py --net ${NET} --cron ${TELEMETRY_KEY} \"${JOB_NAME}\""
+  echo "${NEWCRON}" > /etc/cron.d/dmon
```
Comment (espadolini) [inline install.sh:148]: I can't find a sensible way to do this, but we should see if we can make this update atomic somehow.

File: install.sh [hunk 6]
```diff
@@ -0,0 +1,180 @@
...
+
+RAND=${RANDOM}
+curl -s "${DMON_PY_URL}" > /tmp/${RAND}-dmon.py
+/usr/bin/install -D -o "${USER}" -g "${GROUP}" -m "0755" /tmp/${RAND}-dmon.py "${INSTALL_PATH}/dmon.py"
+rm -f /tmp/${RAND}-dmon.py
+
+
+################### install crontab
+
+if [[ ${USER} == "root" ]]
```
Comment (espadolini) [inline install.sh]: Why not require sudo and use a file in `/etc/cron.d` regardless?

File: install.sh [hunk 7]
```diff
@@ -0,0 +1,180 @@
...
+  # we test for \r and \n because some stuff does \r instead
+  if ! [[ "${c}" == $'\r' || "${c}" == $'\n' ]]
+  then
+    exit 1
+  fi
+}
+
+################### try to figure out default net device
+
+NET=$(ip link show | grep ",UP," | grep -v -E " (veth|br-|dock|lo)" | cut -d ' ' -f 2 | cut -d ':' -f 1 | head -n 1)
```
Comment (espadolini) [inline install.sh]: `ip -oneline link show up` will give you one line per device that's up.

File: install.sh (hunk 3 above)
Comment (drewstreib) [reply install.sh]: removed this comment left over from brew.sh

File: install.sh (hunk 6 above)
Comment (drewstreib) [reply install.sh]: I think I have a notion that some folks might be averse to installing privileged agents and this make it more accessible as an option. They can always just run as root.

File: install.sh (hunk 7 above)
Comment (drewstreib) [reply install.sh]: Done

File: install.sh (hunk 1 above)
Comment (drewstreib) [reply install.sh]: Done re: mktemp. Unsure what to do that is helpful on the rm -f.

Comment (drewstreib) [reply install.sh]: Done.

File: install.sh (hunk 2 above)
Comment (drewstreib) [reply install.sh]: Removed user/group explicit stuff as yes, it was just reading current anyway.

File: install.sh (hunk 4 above)
Comment (drewstreib) [reply install.sh:151]: is `grep -v "/dmon.py "` really likely to accidentally catch something random?

**Pull request merged**